internal/app/app.go

```diff
@@ -104,17 +106,44 @@ func NewListener(c *cli.Context) (*listener.Listener, error) {
 	blockExpiration := c.Duration(blockExpirationFlag.Name)
 	blockKeeper := block.NewRedisBlockKeeper(l, redisClient, maxNumBlocks, blockExpiration)
 
-	maxLen := c.Int64(publisherMaxLenFlag.Name)
-	redisStream := redis.NewStream(redisClient, maxLen)
-
 	topic := c.String(publisherTopicFlag.Name)
-	handler := listener.NewHandler(l, topic, httpEVMClient, blockKeeper, redisStream,
+	publisher, err := getPublisher(c, redisClient, topic)
+	if err != nil {
+		l.Errorw("Fail to get publisher", "error", err)
+
+		return nil, err
+	}
+
+	handler := listener.NewHandler(l, topic, httpEVMClient, blockKeeper, publisher,
 		listener.WithEventLogs(nil, nil))
 
 	return listener.New(l, wsEVMClient, httpEVMClient, handler, sanityEVMClient, sanityCheckInterval,
 		listener.WithEventLogs(nil, nil)), nil
 }
 
+func getPublisher(c *cli.Context, redisClient *redis.Client, topic string) (publisherpkg.Publisher, error) {
```

Comment:
Review note: At first I put the logic below into `NewListener` function, however, it causes `cyclomatic complexity` error when linting. So I separate this logic into its own function.

The parameters passed into this function seem quite counterintuitive to me, though.